core/src/main/scala/com/microsoft/azure/synapse/ml/causal/LinearDMLEstimator.scala

```diff
@@ -0,0 +1,330 @@
+        .setInputCols(Array(SchemaConstants.TreatmentResidualColumn))
+        .setOutputCol("treatmentResidualVec")
+        .setHandleInvalid("skip")
+
+    val treatmentEffectPipelineV1 =
+      new Pipeline().setStages(Array(
+        treatmentModelV1, treatmentResidualTransformer, dropTreatmentPredictedColumnsTransformer,
+        outcomeModelV1, outcomeResidualTransformer, dropOutcomePredictedColumnsTransformer,
+        treatmentResidualVA))
+    val treatmentEffectDFV1 = treatmentEffectPipelineV1.fit(test).transform(test)
+
+    // Step 4 - Cross-fit by fitting the treatment and outcome models with the second split and the residual model with the first split.
+    val treatmentModelV2 = treatmentEstimator.fit(test)
+    val outcomeModelV2 = outcomeEstimator.fit(test)
+
+    val treatmentEffectPipelineV2 =
+      new Pipeline().setStages(Array(
+        treatmentModelV2, treatmentResidualTransformer, dropTreatmentPredictedColumnsTransformer,
+        outcomeModelV2, outcomeResidualTransformer, dropOutcomePredictedColumnsTransformer,
+        treatmentResidualVA))
+    val treatmentEffectDFV2 = treatmentEffectPipelineV2.fit(train).transform(train)
+
+    // Step 5. Average slopes from the two residual models.
+    val regressor = new GeneralizedLinearRegression()
+      .setLabelCol(SchemaConstants.OutcomeResidualColumn)
+      .setFeaturesCol("treatmentResidualVec")
+      .setFamily("gaussian")
+      .setLink("identity")
+      .setFitIntercept(false)
+    val lrmV1 = regressor.fit(treatmentEffectDFV1)
+    val lrmV2 = regressor.fit(treatmentEffectDFV2)
+    val ate = Seq(lrmV1, lrmV2).map(_.coefficients(0)).sum / 2
+
+    ate
+  }
+
+  override def copy(extra: ParamMap): Estimator[LinearDMLModel] = {
+    defaultCopy(extra)
+  }
+
+  @DeveloperApi
+  override def transformSchema(schema: StructType): StructType = {
+    LinearDMLEstimator.validateTransformSchema(schema)
+  }
+}
+
+object LinearDMLEstimator extends ComplexParamsReadable[LinearDMLEstimator] {
+
+  def validateTransformSchema(schema: StructType): StructType = {
+    StructType(schema.fields)
+  }
+}
+
+/** Model produced by [[LinearDMLEstimator]]. */
+class LinearDMLModel(val uid: String)
+  extends Model[LinearDMLModel] with ComplexParamsWritable with Wrappable with BasicLogging {
+  logClass()
+
+  def this() = this(Identifiable.randomUID("LinearDMLModel"))
+
+  val ate = new DoubleParam(this, "ate", "average treatment effect")
+  def getAte: Double = $(ate)
+  def setAte(v: Double): this.type = set(ate, v)
+
+  var ates = new DoubleArrayParam(this, "ates", "treatment effect results for each iteration")
+  def getAtes: Array[Double] = $(ates)
+  def setAtes(v: Array[Double]): this.type = set(ates, v)
+
+  def getCi: Array[Double] = {
+    val ciLowerBound = percentile($(ates), getPercentileLowerBound)
+    val ciUpperBound = percentile($(ates), 100-getPercentileLowerBound)
+    Array(ciLowerBound, ciUpperBound)
+  }
+
+  val percentileLowerBound = new DoubleParam(
```

Comment:
Rename this to "confidenceLevel" with a default value of 0.975, range between 0 and 1. The upper bound can be calculated as {getConfidenceLevel * 100}, and lower bound as {100 - upperBound}.